### PR TITLE
Update custom principal demos for WildFly 29

### PR DESCRIPTION
**Pull request by cam-rod**
- Minimize usage of private Elytron APIs (i.e. replacing implementations of `PrincipalTransformer` with `Function<Principal, Principal>`)
- Remove configuration commands to create default JACC policy
- Update README to match changes
- Bump server to 29.0.0.Final, Core to 21.1.0.Final

### CODE REVIEW

File: custom-principal-ee/application/pom.xml [hunk 1]
```diff
@@ -5,27 +5,22 @@
     <parent>
         <groupId>org.wildfly.security.examples</groupId>
         <artifactId>custom-principal-ee</artifactId>
-        <version>2.0.0.Alpha1-SNAPSHOT</version>
+        <version>2.1.0.Alpha1-SNAPSHOT</version>
```
Comment (fjuma) [inline custom-principal-ee/application/pom.xml:8]: No need to bump this, the main branch is still on 2.0.0.Alpha1. We don't bump the versions for the examples very much.

File: custom-principal-elytron/pom.xml [hunk 2]
```diff
@@ -4,17 +4,17 @@
...
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <maven.compiler.source>11</maven.compiler.source>
         <maven.compiler.target>11</maven.compiler.target>
 
-        <version.org.wildfly.core.wildfly-elytron-integration>20.0.0.Beta7</version.org.wildfly.core.wildfly-elytron-integration>
-        <version.org.wildfly.plugins.wildfly-maven-plugin>4.0.0.Final</version.org.wildfly.plugins.wildfly-maven-plugin>
-        <version.server.bom>27.0.1.Final</version.server.bom>
+        <version.org.wildfly.core.wildfly-elytron-integration>21.1.0.Final</version.org.wildfly.core.wildfly-elytron-integration>
```
Comment (fjuma) [inline custom-principal-elytron/pom.xml:15]: I think this can be removed.

File: custom-principal-ee/pom.xml [hunk 3]
```diff
@@ -4,17 +4,17 @@
...
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <maven.compiler.source>11</maven.compiler.source>
         <maven.compiler.target>11</maven.compiler.target>
 
-        <version.org.wildfly.core.wildfly-elytron-integration>20.0.0.Beta7</version.org.wildfly.core.wildfly-elytron-integration>
-        <version.org.wildfly.plugins.wildfly-maven-plugin>4.0.0.Final</version.org.wildfly.plugins.wildfly-maven-plugin>
-        <version.server.bom>27.0.1.Final</version.server.bom>
+        <version.org.wildfly.core.wildfly-elytron-integration>21.1.0.Final</version.org.wildfly.core.wildfly-elytron-integration>
```
Comment (fjuma) [inline custom-principal-ee/pom.xml:15]: Same here.

File: custom-principal-ejb/pom.xml [hunk 4]
```diff
@@ -14,9 +14,9 @@
 
         <version.org.apache.maven.plugins.maven-ejb-plugin>3.2.1</version.org.apache.maven.plugins.maven-ejb-plugin>
         <version.org.codehaus.mojo.exec-maven-plugin>3.1.0</version.org.codehaus.mojo.exec-maven-plugin>
-        <version.org.wildfly.core.wildfly-elytron-integration>20.0.0.Beta7</version.org.wildfly.core.wildfly-elytron-integration>
-        <version.org.wildfly.plugins.wildfly-maven-plugin>4.0.0.Final</version.org.wildfly.plugins.wildfly-maven-plugin>
-        <version.server.bom>27.0.1.Final</version.server.bom>
+        <version.org.wildfly.core.wildfly-elytron-integration>21.1.0.Final</version.org.wildfly.core.wildfly-elytron-integration>
```
Comment (fjuma) [inline custom-principal-ejb/pom.xml:17]: Same here.

File: custom-principal-ee/README.md [hunk 5]
```diff
@@ -65,11 +65,9 @@ module, which contains our custom principal:
 
 1. The `custom-principal-ee-components` archive is added as a module to the server. This allows the
    custom components to be used by the server.
-2. A default JACC (Jakarta Authorization) policy is created, allowing deployments to use
```
Comment (fjuma) [inline custom-principal-ee/README.md:68]: Just to check, why was this removed?

Comment (cam-rod) [reply custom-principal-ee/README.md:68]: The JACC policy is present by default, so I removed it from the script and didn't see a reason to leave that line in.

Comment (fjuma) [reply custom-principal-ee/README.md:68]: Thanks.

File: custom-principal-ee/application/pom.xml [hunk 6]
```diff
@@ -10,22 +10,17 @@
     </parent>
 
     <artifactId>custom-principal-ee-application</artifactId>
-    <version>2.0.0.Alpha1-SNAPSHOT</version>
```
Comment (fjuma) [inline custom-principal-ee/application/pom.xml:13]: We should keep the existing version there (we just don't need to bump it)

File: custom-principal-ee/components/pom.xml [hunk 7]
```diff
@@ -10,16 +10,11 @@
     </parent>
 
     <artifactId>custom-principal-ee-components</artifactId>
-    <version>2.0.0.Alpha1-SNAPSHOT</version>
```
Comment (fjuma) [inline custom-principal-ee/components/pom.xml:13]: Same here

File: custom-principal-elytron/components/pom.xml [hunk 8]
```diff
@@ -10,16 +10,11 @@
     </parent>
 
     <artifactId>custom-principal-elytron-components</artifactId>
-    <version>2.0.0.Alpha1-SNAPSHOT</version>
```
Comment (fjuma) [inline custom-principal-elytron/components/pom.xml:13]: same here

File: custom-principal-elytron/application/pom.xml [hunk 9]
```diff
@@ -10,11 +10,10 @@
     </parent>
 
     <artifactId>custom-principal-elytron-application</artifactId>
-    <version>2.0.0.Alpha1-SNAPSHOT</version>
```
Comment (fjuma) [inline custom-principal-elytron/application/pom.xml:13]: Same here

**Review by fjuma — APPROVED** [review]
Thanks @cam-rod!
